src/lava/proc/conv/utils.py

```diff
@@ -0,0 +1,274 @@
+    else:
+        raise Exception(
+            f'Expected {name} to be two dimensional.'
+            f'Found {name} = {shape}.'
+        )
+
+
+def signed_clamp(x, bits):
+    """clamps as if input is a signed value within the precision of bits.
+
+    Parameters
+    ----------
+    x : int, float, np array
+        input
+    bits : int
+        number of bits for the variable
+
+    Returns
+    -------
+    same type as x
+        clamped value
+    """
+    base = 1 << bits
+    return (x + base // 2) % base - base // 2  # signed value clamping
+
+
+def output_shape(
+    input_shape, out_channels, kernel_size, stride, padding, dilation
+):
+    """Calculates the output shape of convolution operation.
+
+    Parameters
+    ----------
+    input_shape : 3 element tuple, list, or array
+        shape of input to convolution in XYZ/WHC format.
+    out_channels : int
+        number of output channels.
+    kernel_size : 2 element tuple, list, or array
+        convolution kernel size in XY/WH format.
+    stride : 2 element tuple, list, or array
+        convolution stride in XY/WH format.
+    padding : 2 element tuple, list, or array
+        convolution padding in XY/WH format.
+    dilation : 2 element tuple, list, or array
+        dilation of convolution kernel in XY/WH format.
+
+    Returns
+    -------
+    tuple of 3 ints
+        shape of convolution output in XYZ/WHC format.
+
+    Raises
+    ------
+    Exception
+        for invalid x convolution dimension.
+    Exception
+        for invalid y convolution dimension.
+    """
+    x_out = np.floor(
+        (
+            input_shape[0] + 2 * padding[0]
+            - dilation[0] * (kernel_size[0] - 1) - 1
+        ) / stride[0] + 1
+    ).astype(int)
+    y_out = np.floor(
+        (
+            input_shape[1] + 2 * padding[1]
+            - dilation[1] * (kernel_size[1] - 1) - 1
+        ) / stride[1] + 1
+    ).astype(int)
+
+    if x_out < 1:
+        print(f'{input_shape=}')
+        print(f'{out_channels=}')
+        print(f'{kernel_size=}')
+        print(f'{stride=}')
+        print(f'{padding=}')
+        print(f'{dilation=}')
+        print(x_out, y_out, out_channels)
+        raise Exception(
+            f'Found output x dimension (={x_out}) to be less than 1.'
+            f'Check your convolution sizes.'
+        )
+    if y_out < 1:
+        print(f'{input_shape=}')
+        print(f'{out_channels=}')
+        print(f'{kernel_size=}')
+        print(f'{stride=}')
+        print(f'{padding=}')
+        print(f'{dilation=}')
+        print(x_out, y_out, out_channels)
+        raise Exception(
+            f'Found output y dimension (={y_out}) to be less than 1.'
+            f'Check your convolution sizes.'
+        )
+
+    return x_out, y_out, out_channels
+
+
+# TODO: This might be needed later. The reverse mapping is ambiguous though
+# def input_shape(
+#     output_shape, in_channels, kernel_size, stride, padding, dilation
+# ):
+#     pass
```

Comment:
Remove for release?

Reply:
If we remove for release, my concern is that it might get lost along the way later.

Reply:
You could create an issue instead